xbmc/filesystem/SmartPlaylistDirectory.cpp

```diff
@@ -233,6 +233,12 @@ namespace XFILE
         if (!videoUrl.FromString(baseDir))
           return false;
 
+        // adjust the group in case we're retrieving a grouped playlist
+        // based on artists. This is needed because the video library
+        // is using the actorslink table for artists.
+        if (isGrouped && group == "artists")
+          mvidPlaylist.SetGroup("actors");
```

Comment:
Not entirely sure what you mean with local `group` variable. Do you mean replacing it in the `.xsp` ?